packages/openapi-ts/src/plugins/@pinia-colada/sdk/types.d.ts

```diff
@@ -0,0 +1,51 @@
+import type { IR } from '../../../ir/types';
+import type { Plugin } from '../../types';
+
+export interface QueryHooks {
+  /**
+   * Customize the query key for a specific operation
+   */
+  getQueryKey?: (operation: IR.OperationObject) => string[];
+
+  /**
+   * Override the default query detection logic for an operation.
+   * Return true to force treat as query, false to force treat as mutation.
+   * Return undefined to use default detection logic.
+   */
+  isQuery?: (operation: IR.OperationObject) => boolean | undefined;
+
+  /**
+   * Called when a mutation is created. Return false to skip generating this mutation.
+   */
+  onMutation?: (operation: IR.OperationObject) => boolean | void;
+
+  /**
+   * Called when a query is created. Return false to skip generating this query.
+   */
+  onQuery?: (operation: IR.OperationObject) => boolean | void;
+}
+
+export interface Config extends Plugin.Name<'@pinia-colada/sdk'> {
+  /**
+   * Export queries/mutations from index?
+   * @default true
+   */
+  exportFromIndex?: boolean;
+
+  /**
+   * Group queries by tag into separate files?
+   * @default false
+   */
+  groupByTag?: boolean;
```

Comment:
Can you explain the reasoning behind offering this functionality?

Reply:
It a preference that wasn't available with the current `sdk` plugin. It's very Vue in style though; it allows for a decent amount of lazy-loading and/or treeshaking if you're following the standard Vue import patterns, just importing for the contexts you need in individual Vue SFC files.
You can still import the root index file to get everything in one go, but can progressively add narrower import contexts to components to cut down on bundled component sizes.

Reply:
Thank you for the context! I'd like to see this in snapshot tests, it will be way easier to reason through it. We will most likely want to add this to SDKs too, it's just a matter of priority

Reply:
Not sure how to make the output of the plugin be digestible by the core sdk plugin if that's possible?